.github/workflows/ci.yml

```diff
@@ -455,8 +617,74 @@ jobs:
+          git checkout tags/v${{ matrix.cling-version }}
+          cd ..
+          git clone --depth=1 -b cling-llvm${{ matrix.clang-runtime }} https://github.com/root-project/llvm-project.git
+        else # repl
+          git clone --depth=1 -b release/${{ matrix.clang-runtime }}.x https://github.com/llvm/llvm-project.git
+        fi
+        cd llvm-project
+        # Build
+        mkdir build
+        if [[ "${cling_on}" == "ON" ]]; then
+          cd build
+          emcmake cmake -DLLVM_EXTERNAL_PROJECTS=cling \
+                        -DLLVM_EXTERNAL_CLING_SOURCE_DIR=../../cling  \
+                        -DCMAKE_BUILD_TYPE=Release \
+                        -DLLVM_HOST_TRIPLE=wasm32-unknown-emscripten \
+                        -DLLVM_ENABLE_ASSERTIONS=ON                        \
+                        -DLLVM_TARGETS_TO_BUILD="${{ matrix.llvm_targets_to_build }}" \
+                        -DLLVM_ENABLE_LIBEDIT=OFF \
+                        -DLLVM_ENABLE_PROJECTS="${{ matrix.llvm_enable_projects }}" \
+                        -DLLVM_ENABLE_ZSTD=OFF \
+                        -DLLVM_ENABLE_LIBXML2=OFF \
+                        -DCLANG_ENABLE_STATIC_ANALYZER=OFF \
+                        -DCLANG_ENABLE_ARCMT=OFF \
+                        -DCLANG_ENABLE_BOOTSTRAP=OFF \
+                        -DCMAKE_CXX_FLAGS="-Dwait4=__syscall_wait4" \
+                         ../llvm
+          emmake make clang -j ${{ env.ncpus }}
+          emmake make  cling -j ${{ env.ncpus }}
+          # Now build gtest.a and gtest_main for CppInterOp to run its tests.
+          emmake make gtest_main -j ${{ env.ncpus }}
+        else
+          # Apply patches
+          llvm_vers=$(echo "${{ matrix.clang-runtime }}" | tr '[:lower:]' '[:upper:]')
+          if [[ "${llvm_vers}" == "19" ]]; then
+            git apply -v ../patches/llvm/emscripten-clang${{ matrix.clang-runtime }}-*.patch
+            echo "Apply emscripten-clang${{ matrix.clang-runtime }}-*.patch patches:"
+          fi
+          cd build
+          emcmake cmake -DCMAKE_BUILD_TYPE=Release \
+                        -DLLVM_HOST_TRIPLE=wasm32-unknown-emscripten \
+                        -DLLVM_ENABLE_ASSERTIONS=ON                        \
+                        -DLLVM_TARGETS_TO_BUILD="${{ matrix.llvm_targets_to_build }}" \
+                        -DLLVM_ENABLE_LIBEDIT=OFF \
+                        -DLLVM_ENABLE_PROJECTS="${{ matrix.llvm_enable_projects }}" \
+                        -DLLVM_ENABLE_ZSTD=OFF \
+                        -DLLVM_ENABLE_LIBXML2=OFF \
+                        -DCLANG_ENABLE_STATIC_ANALYZER=OFF \
+                        -DCLANG_ENABLE_ARCMT=OFF \
+                        -DCLANG_ENABLE_BOOTSTRAP=OFF \
+                        -DCMAKE_CXX_FLAGS="-Dwait4=__syscall_wait4" \
+                        ../llvm
+          emmake make clang -j ${{ env.ncpus }}
+          emmake make clang-repl -j ${{ env.ncpus }}
+          emmake make lld -j ${{ env.ncpus }}
```

Comment:
This looks good. Couple small concerns 

1) I would really like if we could do 
```
-DCMAKE_TOOLCHAIN_FILE=path_to_/emscripten/cmake/Modules/Platform/Emscripten.cmake
````

It is added by default on emscripten-forge. So if you see the description behind this file (maybe on github or if you have emsdk installed locally).. it says this 
```
# This file is a 'toolchain description file' for CMake.
# It teaches CMake about the Emscripten compiler, so that CMake can generate
# makefiles from CMakeLists.txt that invoke emcc.
```
So it basically makes any wasm build more configured towards emscripten.

2) I am guessing emmake make install or something is enough ?

Reply:
3. Also I think I pointed this above, there are many things that an emscripten build for llvm would generate that would just not be of any use to us (not for xeus-cpp-lite and even for other projects like clad etc etc)
Some options like these can be used
```
    -DLLVM_INCLUDE_BENCHMARKS=OFF                   \
    -DLLVM_INCLUDE_EXAMPLES=OFF                     \
    -DLLVM_INCLUDE_TESTS=OFF                        \
    LLVM_INCLUDE_DOCS=OFF 
    
```
Also I don't remember why but for sure (this change was made last year) but there was some important reason why Thorsten (the main maintainer at emscripten-forge) had added 
```
-DLLVM_ENABLE_THREADS=OFF                       \
```
to the recipe on emscripten-forge

Reply:
@anutosh491 If you see here https://github.com/compiler-research/CppInterOp/actions/runs/12274727581/job/34248254421#step:24:255 `-DCMAKE_TOOLCHAIN_FILE=path_to_/emscripten/cmake/Modules/Platform/Emscripten.cmake` is already added. That is one of the things ecmake does (i.e. add the necessary flags). To address your other points the reason I avoid the other cmake flags is that because I only build certain targets I avoid building the docs (tests are off by default). It also allows us to build faster since I hit ram issues which limited me to 2 core builds due to one of the above options. The only one I think we could benefit from in the ci is the threads one.